clang/test/Driver/modules.m

```diff
@@ -57,6 +57,14 @@
 // CHECK-MODULE-MAP-FILES: "-fmodule-map-file=foo.map"
 // CHECK-MODULE-MAP-FILES: "-fmodule-map-file=bar.map"
 
+// RUN: %clang -fno-modules -fmodule-name=foo -c -### %s 2>&1 | FileCheck -check-prefix=CHECK-PROPAGATE-MODULE-NAME %s
```

Comment:
Add some comment about the behavior?

Reply:
Done.

Reply:
If the comment is about layering check, we should add
`-fmodules-decluse -fmodule-map-file=...` as well. We have some tests for C/C++ but not for ObjC.

Reply:
I changed the tests to directly test for layering check, since that's what I care about.  I don't see analogous tests for c/c++.  Should I add them?